05-data-structures-part2.Rmd

```diff
@@ -330,44 +423,54 @@ Lets run a basic linear regression on the gapminder dataset:
 
 ```{r}
 # What is the relationship between life expectancy and year?
-l1 <- lm(lifeExp ~ year, data=gapminder)
+reg <- lm(lifeExp ~ year, data=gapminder)
 ```
 
-We won't go into too much detail of what I just wrote, but briefly;
-the `~` denotes a formula, which means treat the variable on the left of the
-`~` as the left hand side of the equation (or response in this case), and
-everything on the right as the right hand side. By telling the linear
-model function to use the gapminder data frame, it knows to look for those
-variable names as its columns.
+We won't go into too much detail, but briefly: 
+
+* `lm` estimates linear statistical models
+* The first argument is a formula, with  `a ~ b` meaning that `a`,
+     the dependent (or response) variable, is a
+    function of `b`, the independent variable. 
+* We tell `lm` to use the gapminder data frame, so it knows where to
+ find the variables `lifeExp` and `year`. 
+
+
 
 Let's look at the output:
 
 ```{r}
-l1
+reg
 ```
 
 Not much there right? But if we look at the structure...
 
 ```{r}
-str(l1)
+str(reg)
 ```
 
-There's a lot of stuff, stored in nested lists! This is why the structure
-function is really useful, it allows you to see all the data available to
-you returned by a function.
+There's a great deal stored in nested lists! The structure function
+allows you to see all the data available, in this case, the data that
+was returned by the `lm` function.
 
 For now, we can look at the `summary`:
 
 ```{r}
-summary(l1)
+summary(reg)
 ```
 
 As you might expect, life expectancy has slowly been increasing over
 time, so we see a significant positive association!
 
+> #### Challenge 6 {.challenge}
+>
+> Write this function: `lm(lifeExp ~ year, data=gapminder)` without
+> using the `data` argument. Verify that you get the same answer.
```

Comment:
On Sat, Aug 29, 2015 at 1:54 PM, Naupaka Zimmerman <notifications@github.com
> wrote:

> In 05-data-structures-part2.Rmd
> <https://github.com/swcarpentry/r-novice-gapminder/pull/59#discussion_r38264827>
> :
>
> >  ```
> >
> >  As you might expect, life expectancy has slowly been increasing over
> >  time, so we see a significant positive association!
> >
> > +> #### Challenge 6 {.challenge}
> > +>
> > +> Write this function: `lm(lifeExp ~ year, data=gapminder)` without
> > +> using the `data` argument. Verify that you get the same answer.
>
> This is a *bad idea* because it will mess up later attempts to use the
> model for prediction. I would take this specific challenge out.
>
>
> It's no problem (and no great loss) to remove this Challenge, but I'm not
understanding the problem:

library(gapminder)
gm1 <- lm(lifeExp ~ year, data=gapminder)
gm2 <- lm(gapminder$lifeExp ~ gapminder$year)
all.equal(predict(gm1), predict(gm2))

So probably I'm missing something...